act-rules-format/act-rules-format.bs

```diff
@@ -433,6 +437,33 @@ Even concepts like headings and images can be misunderstood. These terms could r
   </blockquote>
 </aside>
 
+<aside class=example>
+  <header>Example subjective applicability for a rule testing that elements styled as a heading use correct heading markup. This rule applicability cannot be written objectively since "styled as a heading" is subjective and depends on the context of the page.</header>
+  <blockquote>
+    <p>The rule applies to any HTML element that is styled as a heading. Elements that are styled as a heading may include features such as a larger font-size than nearby text, bolding or changing of the font, or the use of whitespace or shapes that visually distinguish the element text.</p>
+  </blockquote>
+</aside>
+
+<aside class=example>
+  <header><strong>Non-conforming Example:</strong>This example demonstrates an applicability that attempts to be subjective, but does not meet the "must be unambiguous" requirement. For example, non-text content could refer to images, interactive elements, or even a text emoji like <span role="figure" aria-label="Text smiley face created using a colon and parenthesis">:)</span>. Due to the ambiguity in the applicability it is impossible to tell.</header>
+  <blockquote>
+    <p>Applicability: This rule applies to any non-text content.</p>
+    <p>Expectation: Each test target has an accessible name.</p>
+  </blockquote>
+</aside>
+
+<aside class=example>
+  <header><strong>Non-conforming Example:</strong> While an objective applicability should be used when possible, care should be taken to ensure parts of the applicability are not included in other sections of the rules format such as the expectation. Below is an <strong>incorrect</strong> example of using an objective applicability, where the phrase, ".... except if the test target is part of a text node that is purely decorative or does not express anything in human language" in the expectation should be included in the applicability. When fixed, the full applicability would be, "This rule applies to any visible character in a text node that is a child in the flat tree of an HTML element, except if the test target is part of a text node that is purely decorative or does not express anything in human language."</header>
+  <blockquote>
+    <p>Applicability: This rule applies to any visible character in a text node that is a child in the flat tree of an HTML element </p>
+    <p>Expectation: For each test target, the highest possible contrast between the foreground colors and background colors is at least 3.0:1 for large scale text and 4.5:1 for other texts, except if the test target is part of a text node that is purely decorative or does not express anything in human language.</p>
+  </blockquote>
```

Comment:
**Issue:** From a rather logical/specification point of view, there is nothing wrong with that rule. It is only "wrong" because we know that the corresponding SC doesn't apply to non-text content and therefore the exception should be in the Applicability.
But if we change the Expectation to "either the target has good contrast, or it is non-text content", it suddenly doesn't look that bad. And for some other SC, we have "either the text size can be changed, or size is essential" where the essential size is subjective and but is natural in the Expectation.

From a more abstract point of view, consider:
* Applicability: (something objective), except if (something subjective)
  Expectation: (something objective)
* Applicability: (something objective)
  Expectation: (something objective) or (something subjective)
* Applicability: (something objective)
  Expectation: (something objective), except if (something subjective)

All three are doing the same thing, and depending on the Accessibility Requirement, could be legit. So, I'm uneasy about saying that this example is "bad" based on, well, mostly subjective criteria 😉
(I do agree it is bad and cheating around objectivity by moving part of the Applicability in the Expectation; I do not feel we can really define why it's bad in terms objective enough for a specification).

---
Additionally, the normative part of the text says nothing about "do not cheat by moving stuff in the Expectation", which makes this example come quite out of the blue. The previous example illustrate the "ambiguous" vs "subjective" difference that is mentioned earlier in the text, but this example looks out of place (unless you have the historical context of ACT rules and know we did that trick for a long time while format v1.0 was still a thing).

Reply:
I would disagree that the three abstract forms you show up there result in the same behavior. Specifically, we are trying to prevent cases that should be inapplicable showing up as passed (understanding that passed and inapplicable are often treated similarly for us). When you use form number from your comment:

> Applicability: (something objective), except if (something subjective)
> Expectation: (something objective)

The "except if (something subjective)" clause in the applicability will prevent the rule from be evaluated on content that meets the accepted criteria, thus making it inapplicable. 

> Applicability: (something objective)
> Expectation: (something objective) or (something subjective)

> Applicability: (something objective)
> Expectation: (something objective), except if (something subjective)

However, for these second two forms the rule **WILL** be evaluated because the applicability matches, but the expectation will PASS the rule since it is covered by an exception in the Expectation. With that said, the point still remains if we care if one form says inapplicable while the other says passed - we could have a deeper discussion on that point.

I appreciate the thought that this seems to come out of the blue, we could perhaps frame this up better earlier on in the document. My thought process was to show a type of "anti-pattern" that we were hoping to dissuade the use of in the future as we didn't feel it was following the intentions of the format.

Reply:
Yes, they are not equivalent indeed. They report the same failed targets, which is what the rules mostly care about.

My point is more than having this one as a bad practice only stems from the SC it applies to, not from the shape of the expectation itself. That is, the SC is phrased in a way that non-text is "inapplicable", so we want the rule to reflect it. But using "except (some condition)" or even "except (subjective condition)" in the Expectation is not bad per se.

Staying on 1.4.3, consider:

* Applicability: Any character in any text node (as current)
*  Expectation: good contrast, except if non-text or if logotype.

Now, the first exception (non-text) is "bad" because the SC only applies to text. But the second exception (logotype) is "good" because the SC does apply to (text) logotypes and has a special exception in its requirement.

(actually, we may argue whether logotypes should be passing or inapplicable for such a rule…)

From a "find errors" point of view, whether these exceptions are part of the Applicability or Expectation makes no difference. From a formal/logical point of view, there is no real difference between these exceptions. I'm a bit reluctant in phrasing "that" as an anti-pattern without really being able to define what "that" is. It is "trying to sneak the subjective part of Applicability into the Expectation", but that itself feels quite subjective since I find it hard to decide whether something should be Inapplicable or Passing, even more without the link to the requirement the rule should map to.

So, as much as I like this guideline, and it feels like a good one for developers of rules (e.g., in the CG's design document), I'm also not sure it belongs to the rule format itself.

---
I'm also not sure about "forcing" this guideline upon other groups (than the ACT rules CG) who want to maintain a ruleset (e.g., at Siteimprove we have our own ruleset with copies of most rules from the CG, plus several custom rules). Changes we make to the format de facto force other groups to adapt or stop saying their ruleset is following the format. I believe the format benefits from having several rulesets that can notably be used as incubators for rules to be ultimately approved by the AG and included in Understanding WCAG.

Reply:
If I could push the conversation to another rule as an example for a moment. The [Error message describes invalid form field value](https://www.w3.org/WAI/standards-guidelines/act/rules/36b590/proposed/) rule uses the following text in each of its three expectations:

> Each test target either has no [form field error indicators](https://www.w3.org/WAI/standards-guidelines/act/rules/36b590/proposed/#form-field-error-indicator), or at least one of the [form field error indicators](https://www.w3.org/WAI/standards-guidelines/act/rules/36b590/proposed/#form-field-error-indicator) ....

I originally had this as another "anti-pattern" example but removed it since it was duplicative and *I thought* more complex than the color contrast rule above. In this case, I think it is pretty clear that the pushing the applicability into the expectation makes the rule significantly more complex and harder to read. 

So maybe the conversation should be more about when including exceptions in the Expectation vs the Applicability makes more sense. Since they may be both subjective at this point both are a possibility. Here are some reasons to disallow one or the other, i use the term "applicability statements" to mean statements that describe what targets should be tested even when those statements may occur in the expectation.

1. Putting applicability statements in the Expectation vs the Applicability affects the failed test targets -> Applicability statement should go where it makes logical sense and results in the intended outcome for the test targets. 
2. Including exceptions in Expectation or Applicability significantly reduces readability/does not meet the plain language requirement -> Applicability should go in the place that best meets the plain language requirements
3. As a default, applicability statements should always go Applicability unless significant reason for placing it in the Expectation can be show **AND** putting an applicability statement in the expectation does not result in a change of failed test targets.  

I'm not sure I am fully convinced, but I am trying to meet you halfway and find some more formal language we could use to describe the situation.

Reply:
I pretty much agree that "Applicability statements" should go in the Applicability unless there is some very good reason to do otherwise. They are **Applicability** statements, not Expectation statements…

My problem resides in the fact that "Applicability statements" is not really a well defined term. It is at the very least related to the Accessibility requirement the rule is designed for. But ultimately, since the rules are focusing on detecting true errors, it matter little whether something is Passed or Inapplicable. It only matters because we link it back to WCAG and agree that "a page with only non-text content is Inapplicable to 1.4.3, not Passing, so it should be Inapplicable for the ACT rule too". (and in the case of the error indicators, the rule would be more compact and readable if the "no error indicator" bit was factored in the Applicability…)

From the point of view of the ruleset (i.e., the ACT rules CG), it is a good design principle to avoid "Applicability statements" in the Expectation, and we can live with a blurry definition of them. From the point of the rule format, I am not sure we can define "Applicability statement" well enough to enforce (MUST) or even recommend (SHOULD) "no Applicability statement in the expectation".

(plus the fact that the "no applicability statement in the expectation" bit is mentioned in that example without being mentioned earlier in the normative text).

---
Taking another example.

A rule for [2.4.1 Bypass Blocks](https://www.w3.org/TR/WCAG22/#bypass-blocks) could be either Passing or Inapplicable on pages without blocks of repeated contents. Our current rule is Passing them. E.g., [Document has heading for non-repeated content](https://www.w3.org/WAI/standards-guidelines/act/rules/047fe0/proposed/) has Expectation "either there is no [non-repeated content after repeated content](https://www.w3.org/WAI/standards-guidelines/act/rules/047fe0/proposed/#non-repeated-content) or …"

I think this is another "Applicability statement" put in the Expectation for the sake of objectivity of the Applicability. Because the SC doesn't really apply to pages without repeated content.

Now, consider the (PR) best practice rule [First focusable element is link to non-repeated content](https://github.com/act-rules/act-rules.github.io/pull/1539). It doesn't map to any SC. It does relate to technique G1, but that technique isn't really restricted to pages with repeated content.

Currently, that PR also passes pages without repeated content (actually quite specifically with one of its atomic rules!) Is that also an "Applicability statement"? This is much less clear. Telling devs "[applicability] pages should [expectation] either have no repeated content¹, or a skip link" is a decent best practice, as much as "[applicability] pages with repeated content should [expectation] have a skip link". I tend to think that "no repeated content" is an "Applicability statement" here too, but only because the rule indirectly links to 2.4.1 and I use the same reasoning. I.e., a much weaker case than for the full blown "bypass block" rules before (cannot be backed directly to some other normative text). And since having "no repeated content" in the Expectation causes no problem (no false positives), I wouldn't fight to have it moved.

¹ I've seen several customers with "single download" pages that wrap a media file without any other content and thus have no repeated content.

So, in that case, "no repeated content" could be an "applicability statement" or not depending on a lot of outer circumstances. That is why I am uncomfortable with pointing out "no applicability statement in expectation" in the rules format specification.